src/controllers/products.js

```diff
@@ -157,3 +159,39 @@ exports.getProduct = async (req, res, next) => {
     return next(new ErrorResponse('Product not found!', 404));
   }
 };
+
+exports.sellProduct = async (req, res, next) => {
+  const { id, userId } = req.params;
+  const { notes } = req.body;
+  const orderNotes = notes || '';
+  const product = await Product.findById(id);
```

Comment:
The query should check if the product belong to the user (`req.user`), `findOne` is a good choice here.